### PR TITLE
Mint dotnet failure TaskCanceledException

### PR DESCRIPTION
**Pull request by ebozduman**
Fix for sporadic `TaskCanceledException` during Mint test runs.
The added `ConfigureAwait` method is for telling the Task to resume itself on any thread that is available instead of waiting for the thread that originally created it. This will speed up the responses and avoid deadlocks and timeouts.
Deadlocks/Timeouts may happen if another long-running process currently has taken over the thread, async/await was waiting for. If that happens, the awaiting task will be stuck waiting for the thread to complete the other long running task/process, which may end up causing timeouts.
This fix still needs to be tested within Mint environment.

### CODE REVIEW

**Comment by ebozduman** [comment]
I mistakenly picked up changes from others in files:
- Minio.Functional.Tests/Minio.Functional.Tests.csproj 
- Minio/Minio.csproj 
- README.md
I will clean them up and look into the build failure.

**Comment by harshavardhana** [comment]
is this PR ready @ebozduman ?

**Comment by ebozduman** [comment]
Tried to cleanup the `sudo` but changing the permissions on the source file did not help. 

So, it is ready for review now.

File: Minio.Functional.Tests/Program.cs [hunk 1]
```diff
@@ -50,14 +50,7 @@ public static void Main(string[] args)
...
 
         MinioClient minioClient = null;
-
-        var clientHandler = new HttpClientHandler();
-        clientHandler.ServerCertificateCustomValidationCallback = (sender, cert, chain, sslPolicyErrors) =>
-        {
-            return true;
-        };
-        clientHandler.UseProxy = false;
-        var httpClient = new HttpClient(clientHandler);
```
Comment (harshavardhana) [inline Minio.Functional.Tests/Program.cs:60]: did you meant to remove this? @ebozduman

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: Minio.Functional.Tests/Program.cs (hunk 1 above)
Comment (ebozduman) [reply Minio.Functional.Tests/Program.cs:60]: No.This change is supposed to be part of the PR #641.
Thank you for catching it.

**Comment by ebozduman** [comment]
@harshavardhana 
It is ready.

**Comment by harshavardhana** [comment]
Please rebase with master

**Comment by ebozduman** [comment]
Some old code made into this PR. I guess I made some mistake during rebase.
I'll correct those and push it again.

**Comment by ebozduman** [comment]
@harshavardhana 

I think it is ready 'now'.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**